### DISCUSSION TITLE
Couldn't find manifest. [Bug]

### DISCUSSION THREAD

**Discussion by JuanMM94** [ScoopInstaller/Scoop#5083] [Help and Questions] [answered]
#### Current Behavior

```
PS > scoop status

> WARN  Scoop bucket(s) out of date. Run 'scoop update' to get the latest changes.
> firefox   102.0.1                                                            Manifest removed

PS > scoop update

> [redacted]

PS > scoop bucket list

> extras      https://github.com/ScoopInstaller/Extras           08/04/22 05:29:31      1629
> games       https://github.com/Calinou/scoop-games             08/04/22 05:31:06       235
> main        https://github.com/ScoopInstaller/Main             08/04/22 05:30:15      1068
> nerd-fonts  https://github.com/matthewjberger/scoop-nerd-fonts 07/25/22 15:19:02       202
> nirsoft     https://github.com/kodybrown/scoop-nirsoft         06/21/22 13:00:55       271
> nonportable https://github.com/ScoopInstaller/Nonportable      08/02/22 23:47:33        95
> versions    https://github.com/ScoopInstaller/Versions         08/04/22 05:49:26       350

PS > scoop update firefox

> firefox: 102.0.1 (latest version)
> Latest versions for all apps are installed! For more information try 'scoop status'

PS > scoop update firefox --force

> firefox: 102.0.1 -> 102.0.1
> Updating one outdated app:
> ERROR No manifest available for 'firefox'.

PS > scoop checkup

> No problems identified!
```

#### Expected Behavior
It shouldn't show "Manifest removed" as I still see it in its app path (the install.json file)

### System details

**Windows version:** 11

**OS architecture:** x64

**PowerShell version:**

PS > "$($PSVersionTable.PSVersion)"
7.2.5

#### Scoop Configuration

```json
{
  "lastUpdate": "2022-08-04T11:39:28.0193659-03:00",
  "SCOOP_REPO": "https://github.com/ScoopInstaller/Scoop",
  "SCOOP_BRANCH": "master"
}
```

**Comment by lewis-yeung** [comment]
> It shouldn't show "Manifest removed" as I still see it in its app path (the install.json file)

@JuanMM94 The **MANIFEST** file is `manifest.json` in the app path. It must be missing.

**Comment by JuanMM94** [comment]
> > It shouldn't show "Manifest removed" as I still see it in its app path (the install.json file)
> 
> @JuanMM94 The **MANIFEST** file is `manifest.json` in the app path. It must be missing.

Sorry, you're right. It's there, though.

I've found that the typical slink `current` directory isn't routed to the current version. 

![image](https://user-images.githubusercontent.com/66856690/182947190-e4ebd947-08e5-438e-80fe-93a6e87b62a8.png)

Sorry I'm a noob. Might this be what's broken? Why would such a thing happen?

**Comment by JuanMM94** [comment]
> > It shouldn't show "Manifest removed" as I still see it in its app path (the install.json file)
> 
> @JuanMM94 The **MANIFEST** file is `manifest.json` in the app path. It must be missing.

Sorry, you're right. It's there, though.

I've found that the typical slink `current` directory isn't routed to the current version. 

![image](https://user-images.githubusercontent.com/66856690/182947190-e4ebd947-08e5-438e-80fe-93a6e87b62a8.png)

Sorry I'm a noob. Might this be what's broken? Why would such a thing happen?

**Comment by lewis-yeung** [comment]
> Might this be what's broken? Why would such a thing happen?

Not sure, but you can fix it manually.

**Comment by rashil2000** [comment]
Try uninstalling and reinstalling firefox

**Comment by niheaven** [comment, marked as answer]
Just `scoop install firefox` will fix this issue. It will uninstall the broken installation and install it correctly.

**Reply by confused-nerd** [reply]
I just wanted to add, this didn't work for myself:

`scoop install firefox` returned `WARN 'firefox (110.0) is already installed. Use 'scoop update firefox to install a new version.`  running this command:

`scoop update firefox` returned `firefox 110.0 (latest version) Latest versions for all apps are installed! for more information try  'scoop status`

then trying `scoop uninstall firefox` returning validation issues with argument manifest I started searching the issues and in issue #3045 I noted that a [workaround link was under comment](https://github.com/ScoopInstaller/Scoop/issues/3045#issuecomment-493345130) following the same process but for firefox with current version being 112

Not an expert but if someone else is having issues and the answer doesn't work for them it may be useful to have an alternative to try.

TL:DR
- navigated to the scoop extras bucket: `~/scoop/buckets/extras`
- ran `git log --oneline | findstr -i firefox` to find commit id for latest firefox version
- ran `git checkout <commitId> -- .\firefox.json` to checkout the version of firefox.json for the commit from the previous command
- ran `scoop uninstall firefox` to remove the installation. this detected `ERROR 'firefox' isn't installed correctly' and remove the installation and previous versions.
- ran `scoop install firefox` and all was right with the world.

**Reply by HUMORCE** [reply]
Buckets have not been updated correctly, try: `scoop bucket rm extras; scoop bucket add extras` or fix it manually(`cd <extras BUCKET>; git status`), you may also see some errors when `scoop update`.
